internal/driver/driver.go

```diff
@@ -136,23 +165,56 @@ func (s *Driver) HandleWriteCommands(deviceName string, protocols map[string]mod
 		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): deviceResourceName = %v", req.DeviceResourceName))
 		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): protocols: %v, resource: %v, attribute: %v, parameters: %v", protocols, req.DeviceResourceName, req.Attributes, params))
 
-		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
+		// Get the value type from device profile
+		valueType := req.Type
+		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): value type = %v", valueType))
 
+		var value interface{}
+		var err error
+
+		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
 		if key_type_value == "generic" {
-			if value, err := params[i].StringValue(); err == nil {
+			switch valueType {
+			case common.ValueTypeInt8:
+				value, err = params[i].Int8Value()
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): Int8 = %v", value))
+			case common.ValueTypeInt16:
+				value, err = params[i].Int16Value()
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): Int16 = %v", value))
+			case common.ValueTypeString:
+				value, err = params[i].StringValue()
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): String = %v", value))
+			}
+
+			if err == nil {
 				key_timeout_value, _ := cast.ToIntE(req.Attributes["timeout"])
 
 				// initialize the device if it is not initialized already
 				if _, ok := s.generic[deviceLocation]; !ok {
 					s.generic[deviceLocation] = NewUartGeneric(deviceLocation, baudRate, key_timeout_value)
 				}
 
-				// decode the string in hex format
-				txbuf, err := hex.DecodeString(value)
-				if err != nil {
-					return err
+				// Pass the values to the UART end device
+				// Handle data based on the value type mentioned in device profile
+				var txbuf []byte
+				switch valueType {
+				case common.ValueTypeInt8:
+					txbuf = make([]byte, 1)
+					txbuf[0] = byte(value.(int8))
+				case common.ValueTypeInt16:
+					txbuf = make([]byte, 2)
+					// binary.LittleEndian.PutUint16(txbuf, uint16(value.(int16)))
+					binary.BigEndian.PutUint16(txbuf, uint16(value.(int16)))
+				case common.ValueTypeString:
+					// Decode the string to hex format
+					txbuf, err = hex.DecodeString(value.(string))
+					if err != nil {
+						return fmt.Errorf("Driver.HandleWriteCommands(): String decode failed: %v", err)
+					}
 				}
 
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): txbuf = %v", txbuf))
```

Comment:
How will the be displayed for ints? Hex??

Reply:
Hex.

### String:
`level=INFO ts=2023-07-17T19:08:54.471875544Z app=device-uart source=driver.go:222 msg="Driver.HandleWriteCommands(): txbuf = [87 101 108 99 111 109 101 32 116 111 32 116 104 101 32 69 109 98 101 100 100 101 100 32 87 111 114 108 100 33]"`
### Int8:
`level=INFO ts=2023-07-17T19:09:22.838161396Z app=device-uart source=driver.go:222 msg="Driver.HandleWriteCommands(): txbuf = [127]"`
### Int16:
`level=INFO ts=2023-07-17T19:09:40.316190329Z app=device-uart source=driver.go:222 msg="Driver.HandleWriteCommands(): txbuf = [127 255]"`

Reply:
That will work. THX!